### PR TITLE
[script.motr] 19.0.3

### PR DESCRIPTION
**Pull request by large**
### Description
Kodi client addon for connecting to a "Movie On The Run" server, for short: MOTR.

Server can be downloaded here: https://github.com/large/MOTRd/releases
Server is based on .Net and can only be installed in Windows, clients are platform independent.
Server is a "wrapper" around handbreak where clients can queue up conversion of your movies.
Server also serve your files for download or streaming for the clients.

Kodi client uses websocket to communicate with the server, websocket ws4py is included in the package.

Test-server for MOTR can be reached here:
Server: test.motr.pw
Port: 880
PortAlternative: 4444 (with SSL)
Username: test
Password: welcome

If you somehow don't get connection, please do a browser check
http://test.motr.pw:880/
https://test.motr.pw:4444/
(SSL is only self signed, I don't have a valid cert, short on $$$)

I versioned the script to match which kodi release it is for, since it is Matrix only (python 3 only).

Please note that this is my first python script ever, so be kind :)
Also I don't do much PR in Github, so if I made a mistake, sorry in advance!

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.motr/addon.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.motr" name="Movies On The Run" version="19.0.3" provider-name="Lars Werner">
+    <requires>
+        <import addon="xbmc.python" version="3.0.0"/>
+    </requires>
+    <extension point="xbmc.python.script" library="main.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en_GB">Updated for Kodi Matrix!
```
Comment (enen92) [inline script.motr/addon.xml:10]: you have summary and description swapped :)

File: script.motr/resources/lib/kodiutils.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+def show_settings():
+    ADDON.openSettings()
+
+def get_setting(setting):
+    return ADDON.getSetting(setting).strip()
+
+def set_setting(setting, value):
+    ADDON.setSetting(setting, str(value))
+
+def get_setting_as_bool(setting):
```
Comment (enen92) [inline script.motr/resources/lib/kodiutils.py:52]: all these methods don't make any sense anymore. We have getSettingBool, getSettingInt, etc on the python api:
https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html#gaf0205dd5b96354f30a39fbf35599e681

File: script.motr/resources/lib/ws4py/LICENSE [hunk 3]
```diff
@@ -0,0 +1,29 @@
+NOTICE: The version of ws4py included with kaithem has been modified.
```
Comment (enen92) [inline script.motr/resources/lib/ws4py/LICENSE:1]: Sorry this is not allowed, please package ws4py dependency as a script.module and submit as part of another PR.

**Review by enen92 — CHANGES_REQUESTED** [review]

File: script.motr/resources/lib/ws4py/LICENSE (hunk 3 above)
Comment (large) [reply script.motr/resources/lib/ws4py/LICENSE:1]: OK, that was new to me :)
So I'll wrap this as an own module and create a second PR for that then?

Second how do I handle a module import in Kodi, is it the same?
```
from .ws4py import *
from .ws4py.client.threadedclient import WebSocketClient
```

Comment (enen92) [reply script.motr/resources/lib/ws4py/LICENSE:1]: Exactly just the same, just make sure you add it as a dependency in addon.xml

File: script.motr/addon.xml (hunk 1 above)
Comment (large) [reply script.motr/addon.xml:10]: :) Didn't noticed, fixed now, thanks.

File: script.motr/resources/lib/kodiutils.py (hunk 2 above)
Comment (large) [reply script.motr/resources/lib/kodiutils.py:52]: Depending on how you see it. By doing so you have single point for set / get settings.
Makes it much simpler if you want to add some kind of logic later on. Like the get_setting always strip the value it gets.

In my taste for example get_string(id) function is better than each time calling xbmcaddon.Addon().getLocalizedString(id)
And it basically the same thing been called, since ADDON = xbmcaddon.Addon(), then get_string(id) calls ADDON.getLocalizedString(string_id).

I guess this isn't directly wrong so I would like to have it like that, if it is OK for you?

Comment (enen92) [reply script.motr/resources/lib/kodiutils.py:52]: This was just a recommendation, not mandatory to change :)

**Comment by large** [comment]
Well my git skill sucks, I tried to squash it into one PR but failed hard.
Any advice for correct cleanup?

**Comment by enen92** [comment]
Yeah it'll merge when the repo syncs. Please remove the ws4py code from this add-on

File: script.motr/resources/lib/script.py [hunk 4]
```diff
@@ -0,0 +1,732 @@
...
+import os, sys, re, socket, urllib, unicodedata, threading, time, traceback, ssl
+from urllib.request import urlopen
+from resources.lib import kodiutils
+
+import logging
+import xbmc, xbmcgui, xbmcaddon, xbmcvfs
+import datetime
+import gc
+
+try:
```
Comment (enen92) [inline script.motr/resources/lib/script.py:15]: This can be removed, use only Json. Simplejson was only needed in helix :)

File: script.motr/resources/lib/script.py [hunk 5]
```diff
@@ -0,0 +1,732 @@
...
+__pluginname__      = 'script.motr'
+__addon__           = xbmcaddon.Addon(id=__pluginname__)
+__addonid__         = __addon__.getAddonInfo('id')
+__addonname__       = __addon__.getAddonInfo('name')
+__cwd__             = __addon__.getAddonInfo('path')
+__author__          = __addon__.getAddonInfo('author')
+__version__         = __addon__.getAddonInfo('version')
+__language__        = __addon__.getLocalizedString
+__internaldebug__   = False
+__loggerlevel__     = xbmc.LOGINFO
```
Comment (enen92) [inline script.motr/resources/lib/script.py:32]: This has to be changed to logdebug

File: script.motr/resources/skins/Default/720p/convertdialog.xml [hunk 6]
```diff
@@ -0,0 +1,269 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (enen92) [inline script.motr/resources/skins/Default/720p/convertdialog.xml:1]: Can you rename the windows to script-motr-xxxx. Reason is to make sure there is no clash with skin dialogs

File: script.motr/resources/skins/Default/720p/motrwindow.xml [hunk 7]
```diff
@@ -0,0 +1,707 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (enen92) [inline script.motr/resources/skins/Default/720p/motrwindow.xml:1]: Same

**Review by enen92 — CHANGES_REQUESTED** [review]
After these changes and repo sync (~2hours) should be good to go. Thanks for adjusting.

**Comment by large** [comment]
Thank you for the feedback, I have updated as discussed.
Have a nice evening :)

**Review by enen92 — APPROVED** [review]
